### PR TITLE
Add test case for new_PVC scenario of selinux relabeling solution

### PR DESCRIPTION
**Pull request by vkathole**

### CODE REVIEW

File: ocs_ci/helpers/helpers.py [hunk 1]
```diff
@@ -676,6 +676,7 @@ def create_storage_class(
     fs_name=None,
     volume_binding_mode="Immediate",
     allow_volume_expansion=True,
+    kernelMountOptions=None,
```
Comment (PrasadDesala) [inline ocs_ci/helpers/helpers.py:679]: pls also add the doc string for this new arg

File: tests/conftest.py [hunk 2]
```diff
@@ -664,6 +664,7 @@ def factory(
         encryption_kms_id=None,
         volume_binding_mode="Immediate",
         allow_volume_expansion=True,
+        kernelMountOptions=None,
```
Comment (PrasadDesala) [inline tests/conftest.py:667]: pls also add the doc string for this new arg

File: tests/e2e/kcs/test_selinux_relabel_solution.py [hunk 3]
```diff
@@ -330,3 +330,114 @@ def test_selinux_relabel_for_existing_pvc(
         assert (
             pod_restart_time_before_fix > pod_restart_time_after_fix
         ), "Time taken for pod restart after fix is more than before fix."
+
```
Comment (PrasadDesala) [inline tests/e2e/kcs/test_selinux_relabel_solution.py:333]: polarion id, tier markers are missing

File: tests/e2e/kcs/test_selinux_relabel_solution.py [hunk 4]
```diff
@@ -330,3 +330,114 @@ def test_selinux_relabel_for_existing_pvc(
...
+            pvc_factory (function): A call to pvc_factory function
+            service_account_factory (function): A call to service_account_factory function
+            storageclass_factory ((function): A call to storageclass_factory function
+            copies (int): number of copies to write kernel files in pod
+
+        """
+        # Create storageclass with security context
+        self.ocp_project = ocp.OCP(
+            kind=constants.NAMESPACE, namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+        )
```
Comment (PrasadDesala) [inline tests/e2e/kcs/test_selinux_relabel_solution.py:361]: do we need this? calling pvc_factory creates a unique project

File: tests/e2e/kcs/test_selinux_relabel_solution.py [hunk 5]
```diff
@@ -330,3 +330,114 @@ def test_selinux_relabel_for_existing_pvc(
...
+            f' -c "find {data_path} -type f | "shuf" -n {num_of_files}"',
+            timeout=300,
+        )
+        random_files = random_files.split()
+        log.info(f"files are {random_files}")
+        initial_md5sum = []
+        for file_path in random_files:
+            md5sum = res_pod.cal_md5sum(
+                pod_obj=self.pod_obj,
+                file_name=file_path,
```
Comment (PrasadDesala) [inline tests/e2e/kcs/test_selinux_relabel_solution.py:406]: the same code is present even in the first test, maybe you can create a common function and call it in both the tests which reduces the code redundancy

File: tests/e2e/kcs/test_selinux_relabel_solution.py [hunk 6]
```diff
@@ -330,3 +330,114 @@ def test_selinux_relabel_for_existing_pvc(
...
+            )
+            initial_md5sum.append(md5sum)
+
+        # Check data integrity and Get time for pod restart
```
Comment (PrasadDesala) [inline tests/e2e/kcs/test_selinux_relabel_solution.py:410]: change it to 
Delete app pod and ,measure the pod restart time

File: tests/e2e/kcs/test_selinux_relabel_solution.py [hunk 7]
```diff
@@ -330,3 +330,114 @@ def test_selinux_relabel_for_existing_pvc(
...
+        try:
+            wait_for_pods_to_be_running(
+                pod_names=[self.pod_obj.name], timeout=600, sleep=15
+            )
+        except CommandFailed:
+            log.exception(f"Pod {self.pod_obj.name} didn't reach to running state")
+
+        pod_restart_time_after_fix = self.get_pod_start_time(pod_name=self.pod_obj.name)
+        log.info(f"Time taken by pod to restart is {pod_restart_time_after_fix}")
+        final_md5sum = []
```
Comment (PrasadDesala) [inline tests/e2e/kcs/test_selinux_relabel_solution.py:422]: Add a comment - Check Data integrity

File: tests/e2e/kcs/test_selinux_relabel_solution.py [hunk 8]
```diff
@@ -330,3 +330,114 @@ def test_selinux_relabel_for_existing_pvc(
...
+        for file_path in random_files:
+            md5sum = res_pod.cal_md5sum(
+                pod_obj=self.pod_obj,
+                file_name=file_path,
+            )
+            final_md5sum.append(md5sum)
+        assert initial_md5sum == final_md5sum, "Data integrity failed after on new PVC."
```
Comment (PrasadDesala) [inline tests/e2e/kcs/test_selinux_relabel_solution.py:429]: correct the assert message and also print the pvc name for which the integrity check failed -  "Data integrity failed for pvc: {pvc.name}"

File: tests/e2e/kcs/test_selinux_relabel_solution.py [hunk 9]
```diff
@@ -330,3 +330,114 @@ def test_selinux_relabel_for_existing_pvc(
...
+        # Get the node on which application pod is running
+        self.pod_obj = self.get_app_pod_obj()
+        node_name = res_pod.get_pod_node(pod_obj=self.pod_obj).name
+
+        # Check SeLinux Relabeling is set to false
+        log.info("checking for crictl logs")
+        oc_cmd = ocp.OCP(namespace=constants.OPENSHIFT_STORAGE_NAMESPACE)
+        cmd1 = "crictl inspect $(crictl ps --name perf -q)"
+        output = oc_cmd.exec_oc_debug_cmd(node=node_name, cmd_list=[cmd1])
+        key = '"selinuxRelabel": false'
```
Comment (PrasadDesala) [inline tests/e2e/kcs/test_selinux_relabel_solution.py:440]: convert it to a function and replace it in both the previous and the current test

File: tests/e2e/kcs/test_selinux_relabel_solution.py [hunk 10]
```diff
@@ -187,13 +188,55 @@ def get_pod_start_time(self, pod_name):
 
         return None
 
+    def check_selinux_relabeling(self, pod_obj):
+        """
+        Check SeLinux Relabeling is set to false.
+
+        Args:
+            pod_obj (Pod object): App pod
```
Comment (PrasadDesala) [inline tests/e2e/kcs/test_selinux_relabel_solution.py:196]:
```suggestion
            pod_obj (Pod object): App pod
        
```

File: tests/e2e/kcs/test_selinux_relabel_solution.py [hunk 11]
```diff
@@ -187,13 +188,55 @@ def get_pod_start_time(self, pod_name):
 
         return None
 
+    def check_selinux_relabeling(self, pod_obj):
```
Comment (PrasadDesala) [inline tests/e2e/kcs/test_selinux_relabel_solution.py:191]: Please move it some common place outside tests

File: tests/e2e/kcs/test_selinux_relabel_solution.py [hunk 12]
```diff
@@ -187,13 +188,55 @@ def get_pod_start_time(self, pod_name):
...
+
+    def get_random_files(self, pod_obj):
+        """
+        Get random files list.
+
+        Args:
+            pod_obj (Pod object): App pod
+
+        Returns:
+            list : list of random files
```
Comment (PrasadDesala) [inline tests/e2e/kcs/test_selinux_relabel_solution.py:219]:
```suggestion
            list : list of random files
            
```

File: tests/e2e/kcs/test_selinux_relabel_solution.py [hunk 13]
```diff
@@ -330,3 +355,102 @@ def test_selinux_relabel_for_existing_pvc(
...
+        Args:
+            pvc_factory (function): A call to pvc_factory function
+            service_account_factory (function): A call to service_account_factory function
+            storageclass_factory ((function): A call to storageclass_factory function
+            copies (int): number of copies to write kernel files in pod
+
+        """
+        # Create storageclass with security context
+        self.ocp_project = ocp.OCP(
+            kind=constants.NAMESPACE, namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
```
Comment (PrasadDesala) [inline tests/e2e/kcs/test_selinux_relabel_solution.py:387]: `self.ocp_project` this is for storage class? there is no need to provide namespace for storageclass creation

File: tests/e2e/kcs/test_selinux_relabel_solution.py [hunk 14]
```diff
@@ -330,3 +355,102 @@ def test_selinux_relabel_for_existing_pvc(
...
+        )
+
+        # Create PVC using new storageclass
+        self.pvc_obj = pvc_factory(
+            project=self.ocp_project,
+            storageclass=self.storage_class,
+            size="20",
+        )
+        log.info(f"PVC {self.pvc_obj.name} created")
+        teardown_factory(self.pvc_obj)
```
Comment (PrasadDesala) [inline tests/e2e/kcs/test_selinux_relabel_solution.py:403]: you are using pvc_factory for creating PVCs, do you still need to add these PVC objects to the teardown factory here?

File: tests/e2e/kcs/test_selinux_relabel_solution.py (hunk 4 above)
Comment (PrasadDesala) [reply tests/e2e/kcs/test_selinux_relabel_solution.py:388]: why do we need the pvcs to be in only openshift-storage? we cannot apply this solution for the PVCs that are created outside of the openshift-storage namespace?

**Review by PrasadDesala — APPROVED** [review]

**Review by nagendra202 — APPROVED** [review]

**Pull request merged**